src/daemon/XorgDisplayServer.cpp

```diff
@@ -61,7 +61,7 @@ namespace SDDM {
         return QStringLiteral("x11");
     }
 
-    QString XorgDisplayServer::cookie() const {
+    const QByteArray XorgDisplayServer::cookie() const {
```

Comment:
```suggestion
    const QByteArray &XorgDisplayServer::cookie() const {
```

Reply:
`XAuth::cookie()` returns a value. Returning a reference to that would result in a dangling pointer.